cluster/src/main/java/com/hartwig/pipeline/execution/vm/InputDownloadIfBlobExists.java

```diff
@@ -0,0 +1,26 @@
+package com.hartwig.pipeline.execution.vm;
+
+import com.hartwig.pipeline.storage.GoogleStorageLocation;
+
+public class InputDownloadIfBlobExists implements BashCommand {
+
+    private final InputDownload download;
+    private final GoogleStorageLocation googleStorageLocation;
+
+    public InputDownloadIfBlobExists(final GoogleStorageLocation googleStorageLocation) {
+        this.download = new InputDownload(googleStorageLocation);
+        this.googleStorageLocation = googleStorageLocation;
+    }
+
+    @Override
+    public String asBash() {
+        return String.format("gsutil  -q stat  gs://%s/%s; if [ $? == 0 ]; then  %s ; fi",
```

Comment:
Could be shortened to `gsutil -q stat gs://%s/%s && %s` I believe if brevity is important; that's a pretty well-known idiom so I don't think readability would suffer either.